### PR TITLE
WIP - Content Filter - initial commit

### PR DESCRIPTION
**Pull request by duboisp**
See Research 2018-7 in the wet-boew project documentation for more information

### CODE REVIEW

**Review by RobJohnston — COMMENTED** [review]
Hello Pierre:  I hope you find this more useful than annoying.

**Comment by duboisp** [comment]
@Garneauma What in this PR has been added in the tagFilter provisional plugin vs what feature it still need to be added.

**Comment by Garneauma** [comment]
@duboisp I think the Tagfilter plugin eventually got further and further away from this PR as the use case changed. The Tagfilter plugin does not feature a "negative" select for tags, i.e. "!tag", and I don't think it ever will. Same for the "exclusive" select, i.e. "*tag". Also, the Tagfilter plugin does not feature a way to still keep the titles visible when items within its section are hidden, and I don't think we will add that feature.

Basically, I believe the Tagfilter eventually got to be its own plugin which pulled ideas from the Content Filter rather than being an updated version of it.

**Comment by Garneauma** [comment]
Converting this to a draft PR as it will require some more work before it can be merged, if we ever decide it is worth it.
